OpenSim/Simulation/SimbodyEngine/Body.cpp

```diff
@@ -179,43 +180,39 @@ void Body::setInertia(const SimTK::Inertia& inertia)
     upd_inertia()[5] = I[1][2];
 }
 
-//=============================================================================
+//==============================================================================
 // SCALING
-//=============================================================================
-//_____________________________________________________________________________
-/**
- * Scale the body.
- *
- * @param aScaleFactors XYZ scale factors.
- * @param aScaleMass whether or not to scale mass properties
- */
-void Body::scale(const SimTK::Vec3& aScaleFactors, bool aScaleMass)
+//==============================================================================
+void Body::scale(const SimTK::Vec3& scaleFactors, bool scaleMass)
 {
-    Super::scale(aScaleFactors);
-    for(int i=0; i<3; i++) {
-        upd_mass_center()[i] *= aScaleFactors[i];
-    }
+    Super::scale(scaleFactors);
+    upd_mass_center() = get_mass_center().elementwiseMultiply(scaleFactors);
+    scaleInertialProperties(scaleFactors, scaleMass);
```

Comment:
Does this mean that inertial properties get scaled multiple times when `Model::scale()` is called (unnecessarily)? I know that the masses are readjusted if preserveMass is true, but that's not what I'm talking about. If so, this should at least be documented.

Reply:
The `void Body::scale(const SimTK::Vec3& scaleFactors, bool scaleMass)` method is *not* called during `Model::scale()`. This version of `Body::scale()` is (sometimes) used during model construction, e.g., [here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Simulation/Test/testForces.cpp#L197). Seems like a reasonable enough use case to want to leave the method around.

Reply:
Thanks for clarifying.